### PR TITLE
obj_xref: ecdsa support sha3 hash function 

### PR DESCRIPTION
**Pull request by tormath1**
Hi,

In this PR we add the SHA3* support for ECDSA as raised in https://github.com/openssl/openssl/issues/21224. While I tested locally I am wondering if this patch needs a unit test - if yes, any hints regarding the correct location: `test/recipes/15-test_ecparam.t`, `test/recipes/15-test_ecdsa.t` or `test/recipes/25-test_x509.t` ?

Thanks,

Fixes: https://github.com/openssl/openssl/issues/21224

##### Checklist
- [x] tests are added or updated

### CODE REVIEW

**Comment by paulidale** [comment]
I'd suggest adding tests to `test/recipies/30-test_evp_data/evppkey_ecdsa.txt`.  They are required.

I'm not sure if there are any documentation updates required or not.

**Comment by t8m** [comment]
@paulidale I do not think this is something testable by the evp-test. It tests the algorithmic correctness of the ECDSA with SHA3 hash. Which, although useful, I do not think is relevant here. Instead there should be a test that verifies an existing X.509 certificate with these signature algorithms used. That will confirm that the OIDs are present.

So I'd suggest 25-test_verify.t as the place to test with these test certificates added to the test/certs.

**Comment by pmisik** [comment]
I’ve spot issue #21224 accidentally by creating test suite data by using openssl 3.1.2 for another library.
I'm not familiar with OpenSsl test environment.
I took look at 25-test_verify.t.
Do I understand correctly, it is required to add `25-test_verify.t` to test either self-signed certificates by sha3* or|and certificates signed by issuer CA with sha3*?

**Comment by t8m** [comment]
I am not sure the self-signature is being actually verified by the verify command. It would be actually pointless.

**Comment by t8m** [comment]
You need to put the testcase inside one of these sections marked with the `skip "EC is not supported by this OpenSSL build"`

**Comment by pmisik** [comment]
Are you considering adding tests with ecdsa_with_SHA3_224, ecdsa_with_SHA3_256, ecdsa_with_SHA3_512 certificates to have complete coverage of all added OID's?

**Comment by pmisik** [comment]
> You need to put the testcase inside one of these sections marked with the `skip "EC is not supported by this OpenSSL build"` or create a new one.

Do I understand correctly that this means such test are executed just in case EC is supported by compiled OpenSSL build configuration?
Is there a way to check, in web UI, if such test run in CI at least in one configuration?
In web interface i see just:
`25-test_req.t ........................... ok`
but it is unclear which test run and which were skipped.

**Comment by t8m** [comment]
> Is there a way to check, in web UI, if such test run in CI at least in one configuration?

There is no other way than to look at the configure step to see if there is no `no-ec` among the arguments to Configure.

File: test/recipes/25-test_verify.t [hunk 1]
```diff
@@ -350,6 +350,8 @@ SKIP: {
     ok(verify("ee-cert-ec-named-named", "", ["root-cert"],
               ["ca-cert-ec-named"]),
         "accept named curve leaf with named curve intermediate");
+    ok(verify("ee-cert-ec-sha3-384", "", ["root-cert"], ["ca-cert-ec-named"], ),
+        "accept cert generated with EC and SHA3-384");
```
Comment (t8m) [inline test/recipes/25-test_verify.t:354]: Could you please add this to the following section as well - this way the support is verified also with fips+base providers.

**Comment by t8m** [comment]
We will also need a CLA to accept this contribution. Could you please sign it according to: https://www.openssl.org/policies/cla.html

**Comment by t8m** [comment]
> @t8m Thanks for the review and the approval, please note that I sent the CLA yesterday. Should I amend a commit to retrigger the CLA check?

It was not processed yet. It will take a day or two more.

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**